runtime/compiler/env/J2IThunk.hpp

```diff
@@ -73,6 +73,7 @@ class TR_J2IThunkTable
          case 'F': return TC_FLOAT;
          case 'D': return TC_DOUBLE;
          case 'L': return TC_REFERENCE;
+         case 'Q': return TC_VALUE; //unsure about the name
```

Comment:
I would suggest you have `Q` map to `TC_REFERENCE`, and add a TODO to investigate further